File: frontend/src/services/StudentService.js
import axios from 'axios';

// Use relative URL - proxy will forward to backend
const API_BASE_URL = '/api/students';

class StudentService {
  // CREATE - Add a new student
  createStudent(student) {
    return axios.post(API_BASE_URL, student);
  }

  // READ - Get all students
  getAllStudents() {
    return axios.get(API_BASE_URL);
  }

  // READ - Get a student by ID
  getStudentById(id) {
    return axios.get(`${API_BASE_URL}/${id}`);
  }

  // UPDATE - Update student details
  updateStudent(id, student) {
    return axios.put(`${API_BASE_URL}/${id}`, student);
  }

  // DELETE - Remove a student
  deleteStudent(id) {
    return axios.delete(`${API_BASE_URL}/${id}`);
  }
}

export default new StudentService();

File: backend/src/main/java/com/studentrecord/service/StudentService.java
package com.studentrecord.service;

import com.studentrecord.model.Student;
import com.studentrecord.repository.StudentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.Optional;

@Service
public class StudentService {

    @Autowired
    private StudentRepository studentRepository;

    // CREATE - Add a new student
    public Student createStudent(Student student) {
        return studentRepository.save(student);
    }

    // READ - Get all students
    public List<Student> getAllStudents() {
        return studentRepository.findAll();
    }

    // READ - Get a student by ID
    public Optional<Student> getStudentById(Long id) {
        return studentRepository.findById(id);
    }

    // UPDATE - Update student details
    public Student updateStudent(Long id, Student studentDetails) {
        Optional<Student> optionalStudent = studentRepository.findById(id);
        if (optionalStudent.isPresent()) {
            Student student = optionalStudent.get();
            if (studentDetails.getName() != null) {
                student.setName(studentDetails.getName());
            }
            if (studentDetails.getAge() != null) {
                student.setAge(studentDetails.getAge());
            }
            if (studentDetails.getEmail() != null) {
                student.setEmail(studentDetails.getEmail());
            }
            return studentRepository.save(student);
        }
        return null;
    }

    // DELETE - Remove a student
    public void deleteStudent(Long id) {
        studentRepository.deleteById(id);
    }
}
